### ISSUE TITLE
Mac arm64 prebuild

### ISSUE THREAD

**Issue by dgrijuela** [Level/leveldown#766]
I'm not being able to package an electron app for the new mac m1 chip. It says it couldn't find leveldown binaries for that system.

These are the binaries found:

<img width="396" alt="Screen Shot 2021-06-10 at 10 11 34" src="https://user-images.githubusercontent.com/6791788/121489331-47377a80-c9d4-11eb-947a-615286aa08db.png">

Any help? Thanks :)

**Comment by vweevers** [comment]
The ability to produce prebuilt binaries for M1 (Apple silicon) is blocked by https://github.com/actions/virtual-environments/issues/2187.

Are you able to compile from source?

**Comment by vweevers** [comment]
We might be able to cross-compile on an intel mac. Help is wanted, I don't own either.

**Comment by dgrijuela** [comment]
> The ability to produce prebuilt binaries for M1 (Apple silicon) is blocked by [actions/virtual-environments#2187](https://github.com/actions/virtual-environments/issues/2187).
> 
> Are you able to compile from source?

I'll try to ask someone with the M1 do the `npm i leveldown --build-from-source` and post here the results

**Comment by vweevers** [comment]
> We might be able to cross-compile on an intel mac.

An easier option is to build "fat binaries" that work on both x64 and arm64. Here's how `fsevents` did it: https://github.com/fsevents/fsevents/pull/350.

Tasks:

- [ ] Finish https://github.com/Level/leveldown/pull/764
- [ ] Modify `binding.gyp` to build fat binary
- [ ] Test on both M1 and Intel mac (help wanted here) 
- [ ] Find out if these changes affect older mac support (help wanted here) 
- [ ] Figure out how to name and resolve prebuilds. What would the arch be in `darwin-<arch>/node.napi.node`?

**Comment by ralphtheninja** [comment]
I can help with testing on Intel.

**Comment by vweevers** [comment]
Testers wanted! On both Intel and M1. If you have such a machine, please run:

```
git clone --recurse-submodules https://github.com/Level/leveldown.git
cd leveldown
git checkout apple-silicon
npm i --ignore-scripts
npm run prebuild-darwin-x64+arm64
npm run test-prebuild
```

**Comment by ikreymer** [comment]
I can confirm the above steps worked on Apple M1 (updated the package.json to the newly released node-gyp-dep 4.3.0).
I was able to manually add the prebuild to a universal osx electron app and it seems to be working. Thanks!

**Comment by vweevers** [comment]
[6.0.3](https://github.com/Level/leveldown/releases/tag/v6.0.3)
